bin/package.js

```diff
@@ -134,6 +134,37 @@ function buildLinux (cb) {
   electronPackager(Object.assign({}, all, linux), done.bind(null, cb))
 }
 
+function postDarwinism () {
+  var plist = require('plist')
+  var contentsPath = path.join.apply(null, [
+    __dirname,
+    '..',
+    'dist',
+    `${config.APP_NAME}-darwin-x64`,
+    `${config.APP_NAME}.app`,
+    'Contents'
+  ])
+  var resourcesPath = path.join(contentsPath, 'Resources')
+  var infoPlistPath = path.join(contentsPath, 'Info.plist')
+  var webTorrentFileIconPath = path.join.apply(null, [
+    __dirname,
+    '..',
+    'static',
+    'WebTorrentFile.icns'
+  ])
+  var infoPlist = plist.parse(fs.readFileSync(infoPlistPath).toString())
+
+  infoPlist['CFBundleDocumentTypes'] = [{
+    CFBundleTypeExtensions: [ 'torrent' ],
+    CFBundleTypeName: 'BitTorrent Document',
+    CFBundleTypeRole: 'Editor',
+    CFBundleTypeIconFile: 'WebTorrentFile.icns'
+  }]
+
+  fs.writeFileSync(infoPlistPath, plist.build(infoPlist))
+  cp.execSync(`cp ${webTorrentFileIconPath} ${resourcesPath}`)
+}
+
```

Comment:
I decided to make everything concerning osx contained within this function since other platforms wouldn't need `plist` or anything else in here. Still a little weird, but I feel like it's cleaner than what Playback or Atom are doing.